docs(configuration): update infrastructureLogging docs (#4853)
webpack/webpack.js.org: src/content/configuration/other-options.mdx

`object = {}`

### appendOnly

<Badge text='5.31.0+' />

`boolean`

> Only appends lines to the output. Avoids updating existing output e. g. for status messages. This option is only used when no custom [`console`](#console) is provided.

Reviewer: ```suggestion
Append lines to the output instead of updating existing output, useful for status messages. This option is used only when no custom [`console`](#console) is provided.
```

**webpack.config.js**

```javascript
module.exports = {
  //...

>   infrastructureLogging: {
>     appendOnly: true,
>   },

Reviewer: ```suggestion
	infrastructureLogging: {
		appendOnly: false,
		level: "verbose"
	},
	plugins: [
		compiler => {
			const logger = compiler.getInfrastructureLogger("MyPlugin");
			logger.status("first output"); // this line won't be overridden with `appendOnly` enabled
			logger.status("second output");
		}
	]
```

};
```

### colors

<Badge text='5.31.0+' />

`boolean`

> Enable/Disable colorful output. This option is only used when no custom [`console`](#console) is provided.

Reviewer: ```suggestion
Enable colorful output for infrastructure level logging. This option is used only when no custom [`console`](#console) is provided.
```

**webpack.config.js**

```javascript
module.exports = {
  //...

>   infrastructureLogging: {
>     colors: true,
>   },

Reviewer: ```suggestion
infrastructureLogging: {
	colors: true,
	level: "verbose"
},
plugins: [
	compiler => {
		const logger = compiler.getInfrastructureLogger("MyPlugin");
		logger.log("this output will be colorful");
	}
],
```
Would be helpful to add a full example, otherwise it's sort of useless.

};
```

### console

<Badge text='5.31.0+' />

`Console`

> Custom console used for logging.

Reviewer: ```suggestion
Customize the console used for infrastructure level logging.
```

};
```

### stream

<Badge text='5.31.0+' />

`NodeJS.WritableStream = process.stderr`

> Stream used for logging output. Defaults to `process.stderr`. This option is only used when no custom [`console`](#console) is provided.

Reviewer: ```suggestion
Stream used for logging output. Defaults to `process.stderr`. This option is used only when no custom [`console`](#console) is provided.
```
